pootle/apps/pootle_store/unit/filters.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import functools
+
+from django.db.models import Q
+
+from pootle_misc.checks import get_category_id
+from pootle_statistics.models import SubmissionTypes
+
+import pootle_store
+from ..models import SuggestionStates
+
+
+class UnitTextSearch(object):
+    """Search Unit's fields for text strings
+    """
+
+    search_fields = (
+        "source_f", "target_f", "locations",
+        "translator_comment", "developer_comment")
+    search_mappings = {
+        "notes": ["translator_comment", "developer_comment"],
+        "source": ["source_f"],
+        "target": ["target_f"]}
+
+    def __init__(self, qs):
+        self.qs = qs
+
+    def search(self, text, sfields, exact=False):
+        if exact:
+            words = [text]
+        else:
+            words = text.split()
+        result = self.qs.none()
+        for k, v in self.search_mappings.items():
+            if k in sfields:
+                sfields += v
+        for k in filter(lambda x: x in self.search_fields, sfields):
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [map/filter on lambda could be replaced by comprehension](https://www.codacy.com/app/dwayne/pootle/file/1792218155/issues/source?bid=1990595&fileBranchId=2994326#l44)